presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGeneratorContext.java

```diff
@@ -269,7 +269,9 @@ public DruidQueryGenerator.GeneratedDql toQuery()
         }
 
         String expressions = selections.entrySet().stream()
-                .map(s -> s.getValue().getEscapedDefinition())
+                .map(s -> s.getValue().getOrigin() == Origin.TABLE_COLUMN || hiddenColumnSet.contains(s.getKey())
+                        ? s.getValue().getEscapedDefinition()
+                        : s.getValue().getEscapedDefinition() + " AS " + escapeSqlIdentifier(s.getKey().getName()))
```

Comment:
https://github.com/prestodb/presto/blob/381d51ce702533eaee1fd46150f688eafcd0e944/presto-druid/src/main/java/com/facebook/presto/druid/DruidBrokerPageSource.java#L129-L132
Because when a worker reads the response, it gets column data based on the column name.
And when Druid gets expression like `count(*)`, the column name returned is `EXPR$0`..., which does not match the column name generated by presto.
(this worked prior to #15332 because `arrayLines` was used, of course this led to some issue addressed in #15332)

Reply:
Can you help me with where presto generates column names like `sum`, `sum_0`?

Reply:
Presto generates column names in the [variableAllocator](https://github.com/prestodb/presto/blob/4acbaf463937613dc261fe08b9ea0a7a8fcd8cdc/presto-spi/src/main/java/com/facebook/presto/spi/VariableAllocator.java#L33) around [here](https://github.com/prestodb/presto/blob/4acbaf463937613dc261fe08b9ea0a7a8fcd8cdc/presto-spi/src/main/java/com/facebook/presto/spi/VariableAllocator.java#L116).